ui2/src/pages/login/QuickSetup.tsx

```diff
@@ -285,14 +404,17 @@ const QuickSetupTable = (props: {
   async function createRoleContract() {
     const operatorServiceContract = operatorService.contracts[0];
 
-    if (!selectedParty || !operatorServiceContract || selectedServices.length === 0)
-      return undefined;
+    if (!provider || !operatorServiceContract || !services) return undefined;
 
     const id = operatorServiceContract.contractId;
-    const provider = selectedParty.party;
+
+    await ledger.exercise(OperatorService.OfferRegulatorService, id, {
+      provider: operator,
+      customer: provider,
+    });
 
     Promise.all(
-      selectedServices.map(async service => {
+      services.map(async service => {
         if (findExistingOffer(service)) {
           return;
         }
```

Comment:
Should the Regulator service offer be a part of the `findExistingOffer()` and switch case below?

Reply:
the regulator service wont be in the list of services, so it doesn't need to be in the switch case - I could check for existing offers though. Will add that in